tests/autoyast/installation.pm

```diff
@@ -143,6 +142,14 @@ sub run {
     push @needles, 'untrusted-ca-cert' if get_var('SMT_URL');
     # Workaround for removing package error during upgrade
     push(@needles, 'ERROR-removing-package') if get_var("AUTOUPGRADE");
+    # If it's beta, we may match license screen before pop-up shows, so check for pop-up first
+    if (get_var('BETA')) {
+        push(@needles, 'inst-betawarning');
+    }
+    elsif (get_var('AUTOYAST_LICENSE')) {
```

Comment:
I'm not sure about this test variable but are you sure it is the alternative to BETA=1?

Reply:
It's not an alternative, the problem is (or at least was) that license needle was matching even when beta pop-up was there. See L#215. We force the processing order here only. In case of non conflicting needles we could have avoided such logic.